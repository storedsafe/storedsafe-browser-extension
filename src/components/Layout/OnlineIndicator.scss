@import '../../variables';

.online-indicator {
  width: 0.8em;
  height: 0.8em;
  border-radius: 100%;
  transition: $transition-duration*3;
  transition-property: background-color, box-shadow;
  margin: 0.5em;

  &.online {
    background-color: $color-accent-0;
    box-shadow: 0 0 0.2em 0.2em $color-accent-1, 0 0 0.1em 0.1em rgba(0, 0, 0, 0.1) inset;
  }

  &.offline {
    background-color: $color-error-0;
    box-shadow: 0 0 0.2em 0.2em $color-error-1, 0 0 0.1em 0.1em rgba(0, 0, 0, 0.1) inset;
  }
}
