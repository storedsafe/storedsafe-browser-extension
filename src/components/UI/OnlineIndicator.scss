@import '../../variables';

.online-indicator {
  $size: 10px;

  width: $size;
  height: $size;
  border-radius: 100%;
  transition: $transition-duration*3;
  transition-property: background-color, box-shadow;
  margin: 0.5em;

  &.online {
    $shadow: $size / 4;
    background-color: $color-accent-0;
    box-shadow: 0 0 $shadow $shadow $color-accent-1, 0 0 $shadow / 2 $shadow / 2 rgba(0, 0, 0, 0.1) inset;
  }

  &.offline {
    $shadow: $size / 4;
    background-color: $color-danger-0;
    box-shadow: 0 0 $shadow $shadow $color-danger-1, 0 0 $shadow / 2 $shadow / 2 rgba(0, 0, 0, 0.1) inset;
  }
}
