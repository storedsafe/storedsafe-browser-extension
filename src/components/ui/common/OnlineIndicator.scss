@import '../../../variables';

.online-indicator {
  $size: 10px;
  padding: 0.5em;

  .online-indicator-icon {
    border-radius: 100%;
    width: $size;
    height: $size;
    transition: $transition-duration*3;
    transition-property: background-color, box-shadow;
  }

  &.online .online-indicator-icon {
    $shadow: $size / 4;
    background-color: $color-accent-0;
    box-shadow: 0 0 $shadow $shadow $color-accent-1, 0 0 $shadow / 2 $shadow / 2 rgba(0, 0, 0, 0.1) inset;
  }

  &.offline .online-indicator-icon {
    $shadow: $size / 4;
    background-color: $color-danger-0;
    box-shadow: 0 0 $shadow $shadow $color-danger-1, 0 0 $shadow / 2 $shadow / 2 rgba(0, 0, 0, 0.1) inset;
  }
}
