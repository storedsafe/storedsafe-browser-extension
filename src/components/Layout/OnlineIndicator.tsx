import React from 'react';
import './OnlineIndicator.scss';

interface OnlineIndicatorProps {
  online: boolean;
}

export const OnlineIndicator: React.SFC<OnlineIndicatorProps> = ({
  online,
}: OnlineIndicatorProps) => (
  <div className={`online-indicator ${online ? 'online' : 'offline'}`} />
);
