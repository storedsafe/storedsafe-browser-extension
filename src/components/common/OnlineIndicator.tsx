import React from 'react'
import './OnlineIndicator.scss'

interface OnlineIndicatorProps {
  online: boolean
}

export const OnlineIndicator: React.FunctionComponent<OnlineIndicatorProps> = ({
  online
}: OnlineIndicatorProps) => (
  <div className={`online-indicator ${online ? 'online' : 'offline'}`}>
    <div className="online-indicator-icon" />
  </div>
)
