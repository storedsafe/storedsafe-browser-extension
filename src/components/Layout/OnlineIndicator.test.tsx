import * as React from 'react';
import { shallow } from 'enzyme';
import { OnlineIndicator } from './OnlineIndicator';

test('<OnlineIndicator online />', () => {
  const wrapper = shallow(<OnlineIndicator online={true} />);
  expect(wrapper).toMatchSnapshot();
});

test('<OnlineIndicator offline />', () => {
  const wrapper = shallow(<OnlineIndicator online={false} />);
  expect(wrapper).toMatchSnapshot();
});
