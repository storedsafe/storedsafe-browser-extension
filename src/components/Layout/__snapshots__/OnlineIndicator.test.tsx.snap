// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OnlineIndicator offline /> 1`] = `ShallowWrapper {}`;

exports[`<OnlineIndicator online /> 1`] = `ShallowWrapper {}`;
